File: project/app/src/main/java/org/owntracks/android/injection/modules/android/FragmentModules/PlayFragmentModule.java
package org.owntracks.android.injection.modules.android.FragmentModules;


import android.support.v4.app.Fragment;

import org.owntracks.android.injection.scopes.PerFragment;
import org.owntracks.android.ui.welcome.play.PlayFragment;
import org.owntracks.android.ui.welcome.play.PlayFragmentMvvm;
import org.owntracks.android.ui.welcome.play.PlayFragmentViewModel;

import dagger.Binds;
import dagger.Module;

@Module(includes = BaseSupportFragmentModule.class)
public abstract class PlayFragmentModule {

    @Binds
    @PerFragment
    abstract Fragment bindSupportFragment(PlayFragment f);

    @Binds
    @PerFragment
    abstract PlayFragmentMvvm.ViewModel bindViewModel(PlayFragmentViewModel viewModel);
}

File: project/app/src/main/java/org/owntracks/android/ui/welcome/play/PlayFragmentModule.kt
package org.owntracks.android.ui.welcome.play

import androidx.fragment.app.Fragment
import dagger.Binds
import dagger.Module
import org.owntracks.android.injection.scopes.PerFragment

@Module
abstract class PlayFragmentModule {
    @Binds
    @PerFragment
    abstract fun bindSupportFragment(f: PlayFragment?): Fragment?
}